File: GVO Code v3/quad3to3.pas
unit quad3to3;

interface
uses utils, sysutils,windows;
procedure GoQuad3to3(Xcount,Ycount:real);

implementation
uses Main, OMS68SERMC ;

procedure GoQuad3to3(Xcount,Ycount:real);
Var
	CmdStr,CmdStr1, Cmdstr2 : string ;
Begin {3 to 3}
	MovingRa := True;
	MovingDec := True;
	RaPos := xCount ;
	Rapos := Abs(RaPos);
	Str(Rapos:14:0,CmdStr);
	CmdStr := TrimLeft(CmdStr);
	Str(TrkRate,CmdStr2);
	CmdStr1 := 'AX ST;';              { Send RA STOP}
	SendString(CommRecord,pchar(CmdStr));
//
	CmdStr := 'AX  VL'+xvlslew+' MA-'+ CmdStr + ' GD ID; ';
	SendString(CommRecord,pchar(CmdStr));                   {Send RA SLEW}
{}
	DecPos := Ycount;
	Str(Decpos:14:0,CmdStr);
	cmdStr := TrimLeft(cmdStr);
	CmdStr := 'AY VL'+yvlslew+' MA-'+ CmdStr + ' GD ID; ';
	SendString(CommRecord,pchar(CmdStr));                   {Send DEC SLEW}
{Scope is SLEWING}
End;  {3 to 3}

end.

File: GVO Code C++ v1/quad3to3.cpp
#include "quad3to3.h"
#include "GlobalValues.h"
#include "CommUtils.h"

#include <cmath>
#include <iomanip>
#include <sstream>
#include <iostream>
#include <thread>
#include <chrono>

bool GoQuad3to3(double Xcount, double Ycount) {
	movingRA = true;
	movingDEC = true;

	RaPos = std::abs(Xcount);

	std::ostringstream raStream;
	raStream << std::fixed << std::setprecision(0) << RaPos;
	std::string raSteps = raStream.str();

	std::string raCmd = "AX  VL" + xvlslew + " MA-" + raSteps + " GD ID;";

	decPos = std::abs(Ycount);

	std::ostringstream decStream;
	decStream << std::fixed << std::setprecision(0) << decPos;
	std::string decSteps = decStream.str();

	std::string decCmd = "AY  VL" + yvlslew + " MA-" + decSteps + " GD ID;";

	std::cout << "GoQuad3to3:\n";
	std::cout << "  Xcount=" << Xcount << " Ycount=" << Ycount << "\n";
	std::cout << "  RA cmd: [" << raCmd << "]\n";
	std::cout << "  DEC cmd: [" << decCmd << "]\n";

	std::cout << "[GoQuad3to3] before AX send\n";
	bool raOk = SendCommand(raCmd);
	std::cout << "[GoQuad3to3] after AX send, raOk=" << raOk << "\n";

	if (!raOk) {
		std::cout << "Failed to send RA slew command\n";
		return false;
	}

	std::cout << "[GoQuad3to3] before AY send\n";
	bool decOk = SendCommand(decCmd);
	std::cout << "[GoQuad3to3] after AY send, decOk=" << decOk << "\n";

	if (!decOk) {
		std::cout << "Failed to send DEC slew command\n";
		return false;
	}

	return true;
}